pg_analytics/src/hooks/handler.rs

```diff
@@ -1,30 +1,26 @@
 use pgrx::*;
 use std::ffi::{c_char, CString};
 
-use crate::errors::{NotSupported, ParadeError};
+use crate::errors::ParadeError;
 
 static COLUMN_HANDLER: &str = "parquet";
 
-pub trait IsColumn {
+pub trait TableClassifier {
     #[allow(clippy::wrong_self_convention)]
-    unsafe fn is_column(self) -> Result<bool, ParadeError>;
+    unsafe fn table_lists(self) -> Result<(Vec<String>, Vec<String>), ParadeError>;
 }
 
-impl IsColumn for *mut pg_sys::List {
-    unsafe fn is_column(self) -> Result<bool, ParadeError> {
+impl TableClassifier for *mut pg_sys::List {
```

Comment:
nit: Would be cleaner to separate this out into two functions, one that returns a list of row tables, another for column tables.

Reply:
would rather not create two functions and run the same loop twice, but I cleaned up the return format